### PR TITLE
Fix use play lesson

### PR DESCRIPTION
**Pull request by derricklee43**
This took way longer than it should have, lol.

## Summary of Changes ##
- Fixed usePlayLesson hook
- Created played card history
- Added skip functionality 
- Added Replay functionality

https://user-images.githubusercontent.com/39753553/189438846-d828667b-f1af-450f-be37-1329ce2b0825.mov

### CODE REVIEW

File: src/hooks/use-play-lesson.ts [hunk 1]
```diff
@@ -1,46 +1,127 @@
...
   }
 
-  function pauseLesson() {
-    pause();
+  function nextCard(
+    currentCard: LessonCard,
+    upcomingCards: LessonCard[],
+    completedCards: LessonCard[]
+  ) {
+    const newcompletedCards = [currentCard, ...completedCards];
```
Comment (jiejasonliu) [inline src/hooks/use-play-lesson.ts:84]: nit: camel case

File: src/hooks/use-play-lesson.ts [hunk 2]
```diff
@@ -1,46 +1,127 @@
...
+    const newCurrentCard: LessonCard = upcomingCards[0] ?? currentCard;
+    const newUpcomingCards = upcomingCards.length === 0 ? upcomingCards : upcomingCards.slice(1);
```
Comment (jiejasonliu) [inline src/hooks/use-play-lesson.ts:86]: `slice` has a lot of case handling internally so this is equivalent to just `upcomingCards.slice(1)`

from the standard: 
> If start is greater than the index range of the sequence, an empty array is returned.

File: src/models/lazy-audio.ts [hunk 3]
```diff
@@ -14,10 +14,16 @@ export class LazyAudio {
...
 
   public async play(): Promise<void> {
-    return this.loadAudio({
-      onPlayable: () => {
-        this.audio.play();
-      },
+    return new Promise<void>((resolve, reject) => {
+      this.loadAudio({
+        onPlayable: () => {
+          this.audio.addEventListener('ended', () => resolve(), { once: true });
```
Comment (jiejasonliu) [inline src/models/lazy-audio.ts:20]: can you clarify why this is needed?

**Review by jiejasonliu — COMMENTED** [review]
😍

File: src/models/lazy-audio.ts (hunk 3 above)
Comment (derricklee43) [reply src/models/lazy-audio.ts:20]: lol, that was a whoopsies. I was playing with something that didn't work out and I forgot to delete it. 

Good catch, lol

File: src/hooks/use-play-lesson.ts (hunk 2 above)
Comment (derricklee43) [reply src/hooks/use-play-lesson.ts]: oh dope! it should be fixed now

File: src/hooks/use-play-lesson.ts (hunk 1 above)
Comment (derricklee43) [reply src/hooks/use-play-lesson.ts]: lol, that was my bad trying to rename things. fixed

**Review by jiejasonliu — APPROVED** [review]
lgtm, nice work.
fyi cd doesn't work rn bc we are out of memory 🙃

**Pull request merged**